components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIGatewayManager.java

```diff
@@ -671,6 +684,9 @@ private void addEndpoints(API api, APITemplateBuilder builder, GatewayAPIDTO gat
                 if (environment == null) {
                     continue;
                 }
+                if (debugEnabled) {
+                    log.debug(api.getId() + "removing from the environment of " + environment);
```

Comment:
you can't print the environment object. Print the environment name from the for loop.

Reply:
https://github.com/wso2/carbon-apimgt/pull/9019/commits/29a2e4bba591e33476f9fcce64bd8668945123c4